Design/Algorithm_ease_of_use.md

```diff
@@ -0,0 +1,88 @@
+Motivation
+==========
+
+Currently code (especially workflow algorithms) will be full of things
+like
+
+```C
+double tolerance = getProperty("CompressTolerance");
+m_outputEW = boost::dynamic_pointer_cast<EventWorkspace>(m_outputW);
+if ((m_outputEW) && (tolerance > 0.)) {
+  g_log.information() << "running CompressEvents(Tolerance=" << tolerance
+                      << ")\n";
+  API::IAlgorithm_sptr compressAlg = createChildAlgorithm("CompressEvents");
+  compressAlg->setProperty("InputWorkspace", m_outputEW);
```

Comment:
I think you're nearly there with "Alternative 2". You clearly need to call the non-members from the algorithm methods `execAsChild` or whatever. At that point, properties should be expanded out so that the typed values of each property can be passed as arguments. I'm not sure the best way to handle return types.